rules/go/lang/log_output_neutralization.yml

```diff
@@ -0,0 +1,151 @@
+imports:
+  - go_shared_lang_dynamic_request_input
+patterns:
+  - pattern: |
+      $<CALLER>.$<METHOD>($<INPUT>);
+    filters:
+      - either:
+          - variable: CALLER
+            detection: go_lang_log_output_neutralization_logger
+          - variable: CALLER
+            detection: go_lang_log_output_neutralization_zerolog
+      - variable: METHOD
+        regex: \A(Fatal|Panic|Print)(f|ln)?\z
+      - variable: INPUT
+        detection: go_lang_log_output_neutralization_input
+  - pattern: |
+      $<ZEROLOG>.$<EVENT>.$<METHOD>($<INPUT>);
+    filters:
+      - variable: ZEROLOG
+        detection: go_lang_log_output_neutralization_zerolog
+      - variable: EVENT
+        regex: \A(Info|Debug|Error|Trace|Fatal|Panic|Warn)\z
+      - variable: METHOD
+        values:
+          - Msg
+          - Msgf
+      - variable: INPUT
+        detection: go_lang_log_output_neutralization_input
+  - pattern: |
+      $<CALLER>.$<METHOD>($<INPUT>);
+    filters:
+      - either:
+          - variable: CALLER
+            detection: go_lang_log_output_neutralization_logrus
+          - variable: CALLER
+            detection: go_lang_log_output_neutralization_seelog
+      - variable: METHOD
+        regex: \A(WithFields\.)?(Info|Debug|Error|Trace|Fatal|Panic|Warn)\z
+      - variable: INPUT
+        detection: go_lang_log_output_neutralization_input
+  - pattern: |
+      $<CALLER>.$<METHOD>($<INPUT>);
+    filters:
+      - variable: CALLER
+        detection: go_lang_log_output_neutralization_glog
+      - variable: METHOD
+        regex: \A(Info|Warning|Error|Fatal)(Contex)?(Depth)?(f)?\z
+      - variable: INPUT
+        detection: go_lang_log_output_neutralization_input
+  - pattern: |
+      $<ZAP>.$<METHOD>($<INPUT>);
+    filters:
+      - variable: ZAP
+        detection: go_lang_log_output_neutralization_zap
+      - variable: METHOD
+        regex: \A(WithFields\.)?(Info|Log|Error|Fatal|DPanic|Warn)\z
+      - variable: INPUT
+        detection: go_lang_log_output_neutralization_input
+auxiliary:
+  - id: go_lang_log_output_neutralization_logger
```

Comment:
Could be a good to separate these logger calls to a shared method - and use it in go_lang_logger rule too. Happy to do it as part of this PR if folk think it's worthwhile at this stage.

Reply:
Can be done later IMO